### PR TITLE
Add comment to ops bound at python layer

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30420 Comment on a set of ops bound at the python layer
* **#30419 Add comment to ops bound at python layer**

### CODE REVIEW

File: tools/autograd/templates/python_torch_functions.cpp [hunk 1]
```diff
@@ -413,6 +413,9 @@ static PyObject * TypeError_to_NotImplemented_(PyObject* self, PyObject* args, P
 
 ${py_methods}
 
+// XXX: ops that are bound here are not exposed to the C++ api nor the JI2T.
```
Comment (zou3519) [inline tools/autograd/templates/python_torch_functions.cpp]: nit: JI2T -> JIT

File: tools/autograd/templates/python_torch_functions.cpp [hunk 2]
```diff
@@ -413,6 +413,9 @@ static PyObject * TypeError_to_NotImplemented_(PyObject* self, PyObject* args, P
...
+// Any new ops added here should be accompanied with a commment why they are not
```
Comment (zou3519) [inline tools/autograd/templates/python_torch_functions.cpp]: nit: commment -> comment

File: tools/autograd/templates/python_torch_functions.cpp [hunk 3]
```diff
@@ -413,6 +413,9 @@ static PyObject * TypeError_to_NotImplemented_(PyObject* self, PyObject* args, P
...
+// being registered through native_functions.yaml, and include reviwers from C++/JIT
```
Comment (zou3519) [inline tools/autograd/templates/python_torch_functions.cpp]: nit: reviwers -> reviewers

Maybe just write that people should attempt to add the JIT and cpp labels? Contributors might not know who to ping

**Review by zou3519 — APPROVED** [review]
It might be possible to do some "compile time" checking -- we could make it so that PyTorch doesn't compile if someone adds a function here unless they update something (like a count of special-cased functions). We have something similar in [TensorImpl.h](https://github.com/pytorch/pytorch/blob/3990e9d1ca23e39bfa65fe1d907ddb9f9dbf0919/c10/core/TensorImpl.h#L1733-L1736), but this is probably overengineering the problem

**Comment by eellison** [comment]
> It might be possible to do some "compile time" checking -- we could make it so that PyTorch doesn't compile if someone adds a function here unless they update something (like a count of special-cased functions). We have something similar in [TensorImpl.h](https://github.com/pytorch/pytorch/blob/3990e9d1ca23e39bfa65fe1d907ddb9f9dbf0919/c10/core/TensorImpl.h#L1733-L1736), but this is probably overengineering the problem

Yea, that is probably a desirable end result. I'm not sure exactly what that would look like, and I think at the moment we have too many ops that are bound here that we want to remove for it to make sense. Once we get down to a much smaller set of ops that would be nice.

Also thanks for all the spelling corrections lol

**Pull request closed without merging**